src/main/java/com/faforever/api/utils/AuthenticationHelper.java

```diff
@@ -0,0 +1,21 @@
+package com.faforever.api.utils;
+
+import com.faforever.api.data.domain.Player;
+import com.faforever.api.user.FafUserDetails;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.Optional;
+
+/**
+ * @author Dragonfire
+ */
+public class AuthenticationHelper {
+
+  public static Optional<Player> getPlayer(org.springframework.security.core.Authentication auth,
+                                           JpaRepository<Player, Integer> playerRepo) {
+    if (auth != null && auth.getPrincipal() != null && auth.getPrincipal() instanceof FafUserDetails) {
+      return Optional.of(playerRepo.findOne(((FafUserDetails) auth.getPrincipal()).getId()));
+    }
+    return Optional.empty();
+  }
+}
```

Comment:
Remove this helper, use `@AuthenticationPrincipal` instead. If you don't like it, we can create our custom annotation which injects `FafUserDetails`

The helper shouldn't trigger a DB query anyways, since JWT is used which contains all information already. It's already extracted for us - and if we had to fetch it, there's a Spring API to do it properly.